superset-frontend/src/SqlLab/components/TableElement/index.tsx

```diff
@@ -77,6 +77,15 @@ const Fade = styled.div`
 const TableElement = ({ table, actions, ...props }: TableElementProps) => {
   const [sortColumns, setSortColumns] = useState(false);
   const [hovered, setHovered] = useState(false);
+  const [tableNameOverflow, setTableNameOverflow] = useState(false);
+  const tableNameRef = React.useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const element = tableNameRef.current;
+    if (element && element.offsetWidth < element.scrollWidth) {
```

Comment:
you could use optional chaining here instead of the `&&`

Reply:
You caught me, I totally forgot about optional chaining again. I did go in here and try to change this to something like element?.offsetWidth < element?.scrollWidth and apparently typescript doesn't like that. It gives the error element can possibly be undefined and after a quick google, I eventually found my way to TypeScripts docs where it explains why. 

So my intention is to not run the less than comparison unless the element exists but TS's issue is that optional chaining doesn't stop comparisons from happening the way that && does. They just return undefined if the object is null and TS doesn't like when you check if undefined is less than undefined, so it throws the error. Here is the release notes where they talk about it [typescript docs](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing)

I don't know maybe I am just not looking hard enough. Any ideas on how to get around this that don't defeat the purpose?

Reply:
Oh, good point on the typescript error. You could coerce the type like `if ((element?.offsetWidth as number) < (element?.scrollWidth as number))` but in this case what you have is more readable.